diff -Naur hybrid-v35_64-nodebug-pcoem-6_30_223_141.orig/src/include/linuxver.h hybrid-v35_64-nodebug-pcoem-6_30_223_141/src/include/linuxver.h
--- hybrid-v35_64-nodebug-pcoem-6_30_223_141.orig/src/include/linuxver.h	2013-08-01 08:52:22.000000000 +0200
+++ hybrid-v35_64-nodebug-pcoem-6_30_223_141/src/include/linuxver.h	2014-07-02 18:36:28.616225432 +0200
@@ -169,8 +169,12 @@
 #define __devexit
 #endif
 #ifndef __devinit
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)
+#define __devinit
+#else
 #define __devinit	__init
 #endif
+#endif
 #ifndef __devinitdata
 #define __devinitdata
 #endif
