--- a/src/wl/sys/wl_cfg80211_hybrid.c	2014-06-26 11:42:08.000000000 +0100
+++ b/src/wl/sys/wl_cfg80211_hybrid.c	2015-03-02 12:37:17.128248174 +0000
@@ -1441,7 +1441,7 @@ wl_cfg80211_get_station(struct wiphy *wi
 		WL_DBG(("Could not get rate (%d)\n", err));
 	} else {
 		rate = dtoh32(rate);
-		sinfo->filled |= STATION_INFO_TX_BITRATE;
+		sinfo->filled |= BIT(NL80211_STA_INFO_TX_BITRATE);
 		sinfo->txrate.legacy = rate * 5;
 		WL_DBG(("Rate %d Mbps\n", (rate / 2)));
 	}
@@ -1454,7 +1454,7 @@ wl_cfg80211_get_station(struct wiphy *wi
 			return err;
 		}
 		rssi = dtoh32(scb_val.val);
-		sinfo->filled |= STATION_INFO_SIGNAL;
+		sinfo->filled |= BIT(NL80211_STA_INFO_SIGNAL);
 		sinfo->signal = rssi;
 		WL_DBG(("RSSI %d dBm\n", rssi));
 	}
